### PR TITLE
feat!: add custom header support

### PR DESCRIPTION
**Pull request by eshanholtz**

### CODE REVIEW

File: src/Twilio/Rest/Authy/V1/Service/Entity/FactorList.php [hunk 1]
```diff
@@ -81,10 +90,10 @@ public function create(string $binding, string $friendlyName, string $factorType
      *                        efficient page size, i.e. min(limit, 1000)
      * @return Stream stream of results
      */
-    public function stream(int $limit = null, $pageSize = null): Stream {
+    public function stream(array $options = [], int $limit = null, $pageSize = null): Stream {
```
Comment (childish-sambino) [inline src/Twilio/Rest/Authy/V1/Service/Entity/FactorList.php:93]: Are the `read/page/stream` the only part of this change that's breaking? Will the generated code samples also be updated?

I think this warrants a migration guide listing which resources this affects and how to update the client code.

File: tests/Twilio/Integration/Wireless/V1/UsageRecordTest.php [hunk 2]
```diff
@@ -26,7 +26,10 @@ public function testReadRequest(): void {
 
         $this->assertRequest(new Request(
             'get',
-            'https://wireless.twilio.com/v1/UsageRecords'
+            'https://wireless.twilio.com/v1/UsageRecords',
```
Comment (childish-sambino) [inline tests/Twilio/Integration/Wireless/V1/UsageRecordTest.php:29]: Are these changes needed?

Comment (eshanholtz) [reply tests/Twilio/Integration/Wireless/V1/UsageRecordTest.php:29]: Not necessarily. I can add additional logic to prevent these changes. I wasn't sure if it mattered since these are just integration tests.

Comment (childish-sambino) [reply tests/Twilio/Integration/Wireless/V1/UsageRecordTest.php:29]: Meh. It's fine as-is.

File: src/Twilio/Rest/Authy/V1/Service/Entity/FactorList.php (hunk 1 above)
Comment (eshanholtz) [reply src/Twilio/Rest/Authy/V1/Service/Entity/FactorList.php:93]: What generated code samples?

Comment (childish-sambino) [reply src/Twilio/Rest/Authy/V1/Service/Entity/FactorList.php:93]: I can't find any examples specific to these changes in the Twilio docs, but the code generator is also used to generate code samples for every API. Let's follow-up offline.
